### ISSUE TITLE
[LifetimeSafety] False-positive for aliases to reference variable

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#169758]
https://godbolt.org/z/cGxozPYWj

```cpp
#include <iostream>
#include <string>
#include <string_view>
#include <vector>

void foo() {
    std::vector<std::string> v;
    std::string_view view;
    for (const auto& s : v) {
        view = s; // warning: 's' does not live long enough.
    }
    (void)view; // note: later used here.
}

void bar(const std::string& s) {
    std::string_view view;
    {
        const std::string& ref = s; // 's' does not live long enough.
        view = ref;
    }
    (void)view; // later used here.
}

```

**Comment by usx95** [comment]
Verified. https://godbolt.org/z/T4dv5aTf3

```cpp
#include <string>
#include <string_view>

std::string_view bar(const std::string& s) { // got suggestion here. ok.
    std::string_view view;
    const std::string& ref = s;
    view = ref;
    return view;
}

std::string_view foo(std::string s) {
    return bar(s); // error. ok. inference also works.
}
```
